### ISSUE TITLE
VMware Tanzu support

### ISSUE THREAD

**Issue by Ankurk99** [kubearmor/KubeArmor#1064]
## Feature Request

**Short Description**

Support for VMware Tanzu needs to be validated. Validation has to be done for:
- [ ] enforcement supported? What is the LSM used?
- [ ] Audit/Observability supported
- [ ] Paste the output of karmor probe

As an FYI, the tasks involved would be:
1. Setup a k8s cluster using [free tier vmware tanzu program](https://tanzu.vmware.com/content/blog/vmware-tanzu-community-edition-announcement) 
2. Install KubeArmor
3. Check sample workloads and verify if following things are working:
   * Policy Enforcement
   * Alerts/Telemetry
4. Get `karmor probe` output reference and attach in this issue
5. Update [Kubearmor support matrix](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/support_matrix.md)

edit: added comments suggestions

**Comment by nyrahul** [comment]
As an FYI, the tasks involved would be:
1. Setup a k8s cluster using [free tier vmware tanzu program](https://tanzu.vmware.com/content/blog/vmware-tanzu-community-edition-announcement) 
2. Install KubeArmor
3. Check sample workloads and verify if following things are working:
   * Policy Enforcement
   * Alerts/Telemetry
4. Get `karmor probe` output reference and attach in this issue
5. Update [Kubearmor support matrix](https://github.com/kubearmor/KubeArmor/blob/main/getting-started/support_matrix.md)

**Comment by saurabhsy8** [comment]
@nyrahul  I want to work on this, please assign it to me

**Comment by saurabh3460** [comment]
@nyrahul I want to work on this, please assign it to me

**Comment by saurabh3460** [comment]
@nyrahul I want to work on this, please assign it to me

**Comment by vrabbi** [comment]
When you say tanzu, there are actually 3 tanzu distributions, so it is important to be exact on which distro this is being tested. There is tkgi (foremerly pks), tkgm (multi cloud tkg on aws azure and vsphere), and tkgs (vsphere with tanzu). Also they support ubuntu and photonOS3 as well as amazon linux 2 when running on aws so important again to be explicit about what is tested and validated

**Comment by Ankurk99** [comment]
@saurabh3460 Were you able to setup the environment? And can you please mention which one are you testing first?

**Comment by saurabh3460** [comment]
@Ankurk99 I struggled with cli installation but did that so far and I am going with amazon and ubuntu setup first

**Comment by jatinagwal** [comment]
@saurabh3460 Can I pick this up ?

**Comment by saurabh3460** [comment]
sorry @jatinagwal for late reply sure take it, @nyrahul can you please assign it to him?

**Comment by HariVamsiK** [comment]
@jatinagwal are you working on this? If not, I want to take this up.

**Comment by jatinagwal** [comment]
> @jatinagwal are you working on this? If not, I want to take this up.
> 

Hey, sorry I mentioned progress on slack. Actually, to deploy TKG i don't have enough resources. You can pick this

**Comment by Ankurk99** [comment]
Hey @HariVamsiK, were you able to setup the Tanzu instance?

**Comment by HariVamsiK** [comment]
No not yet, I'm still figuring out the instances to use to setup the cluster. I'll update you the progress once I'm done setting up. I'll be using AWS for this.

**Comment by Ankurk99** [comment]
Hey @HariVamsiK, Do you require any help in setting up the Tanzu instance?

**Comment by HariVamsiK** [comment]
Yeah I might need it, I talked to @saurabh3460 about this and got some inputs from him.

Sorry I was busy with some academic work. Will start the work soon and report about it in the next community meeting.

**Comment by Ankurk99** [comment]
No worries, let's discuss about that in the community call then. If you need any help till then, please feel free to ask here.

**Comment by akashsawan1** [comment]
hey, If no one is working ?
I would love to work on this.
Please assign it to me.
@Ankurk99

**Comment by HariVamsiK** [comment]
@akashsawan1 I'm working on it, Was able to create the management cluster too. Testing only remains now.

**Comment by akashsawan1** [comment]
> @akashsawan1 I'm working on it, Was able to create the management cluster too. Testing only remains now.

Oh great , let me know if you got stuck somewhere, we can work on resolving the issue together.
